Missing function comments

Please fix golint problems

```
golint pkg/stub/providers/aws.go
pkg/stub/providers/aws.go:13:6: exported type AwsProvider should have comment or be unexported
pkg/stub/providers/aws.go:16:1: exported method AwsProvider.CreateStorageClass should have comment or be unexported
pkg/stub/providers/aws.go:44:1: exported method AwsProvider.GenerateMetadata should have comment or be unexported
pkg/stub/providers/aws.go:73:1: exported method AwsProvider.CheckBucketExistence should have comment or be unexported
pkg/stub/providers/aws.go:77:1: exported method AwsProvider.CreateObjectStoreBucket should have comment or be unexported
......
```

--- a/pkg/apis/banzaicloud/v1alpha1/types.go
+++ b/pkg/apis/banzaicloud/v1alpha1/types.go
@@ -6,6 +6,7 @@ import (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// ObjectStoreList struct for lists
 type ObjectStoreList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata"`
@@ -14,6 +15,7 @@ type ObjectStoreList struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// ObjectStore struct bounds together the related struct
 type ObjectStore struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata"`
@@ -21,9 +23,12 @@ type ObjectStore struct {
 	Status            ObjectStoreStatus `json:"status,omitempty"`
 }
 
+// ObjectStoreSpec struct holds specs
 type ObjectStoreSpec struct {
 	Name string
 }
+
+// ObjectStoreStatus struct holds status related things
 type ObjectStoreStatus struct {
 	// Fill me
 }

--- a/pkg/stub/providers/aws.go
+++ b/pkg/stub/providers/aws.go
@@ -10,9 +10,11 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// AwsProvider holds info about Aws provider and allows us to implement the common interface
 type AwsProvider struct {
 }
 
+// CreateStorageClass creates a StorageClass based on specs described on PVC
 func (aws *AwsProvider) CreateStorageClass(pvc *v1.PersistentVolumeClaim) error {
 	logrus.Info("Creating new storage class")
 	provisioner, err := aws.determineProvisioner(pvc)
@@ -41,10 +43,12 @@ func (aws *AwsProvider) CreateStorageClass(pvc *v1.PersistentVolumeClaim) error 
 	})
 }
 
+// GenerateMetadata generates metadata which are needed to create a StorageClass
 func (aws *AwsProvider) GenerateMetadata() error {
 	return nil
 }
 
+// determineParameters determines the access mode from PVC
 func (aws *AwsProvider) determineParameters(pvc *v1.PersistentVolumeClaim) (map[string]string, error) {
 	//var parameter = map[string]string{}
 	for _, mode := range pvc.Spec.AccessModes {
@@ -56,6 +60,7 @@ func (aws *AwsProvider) determineParameters(pvc *v1.PersistentVolumeClaim) (map[
 	return nil, errors.New("could not determine parameters")
 }
 
+// determineProvisioner determines what kind of provisioner should the storage class use
 func (aws *AwsProvider) determineProvisioner(pvc *v1.PersistentVolumeClaim) (string, error) {
 	for _, mode := range pvc.Spec.AccessModes {
 		switch mode {
@@ -70,10 +75,12 @@ func (aws *AwsProvider) determineProvisioner(pvc *v1.PersistentVolumeClaim) (str
 	return "", errors.New("AccessMode is missing from the PVC")
 }
 
+// CheckBucketExistence checks if the bucket already exists
 func (aws *AwsProvider) CheckBucketExistence(store *v1alpha1.ObjectStore) (bool, error) {
 	return false, nil
 }
 
+// CreateObjectStoreBucket creates a bucket in a cloud specific object store
 func (aws *AwsProvider) CreateObjectStoreBucket(store *v1alpha1.ObjectStore) error {
 	return nil
 }

--- a/pkg/stub/providers/azure.go
+++ b/pkg/stub/providers/azure.go
@@ -17,12 +17,14 @@ import (
 	"net/http"
 )
 
+// Metadata holds info about Azure
 type Metadata struct {
 	location          string
 	subscriptionID    string
 	resourceGroupName string
 }
 
+// AzureProvider holds info about Azure provider and allows us to implement the common interface
 type AzureProvider struct {
 	metadata Metadata
 }
@@ -34,6 +36,7 @@ const (
 	kind           = "kind"
 )
 
+// CreateStorageClass creates a StorageClass based on specs described on PVC
 func (az *AzureProvider) CreateStorageClass(pvc *v1.PersistentVolumeClaim) error {
 	logrus.Info("Creating new storage class")
 	provisioner, err := az.determineProvisioner(pvc)
@@ -65,6 +68,7 @@ func (az *AzureProvider) CreateStorageClass(pvc *v1.PersistentVolumeClaim) error
 	})
 }
 
+// GenerateMetadata generates metadata which are needed to create a StorageClass
 func (az *AzureProvider) GenerateMetadata() error {
 	logrus.Infof("Getting Metadata from service")
 	var metadatas = [3]string{
@@ -99,6 +103,7 @@ func (az *AzureProvider) GenerateMetadata() error {
 	return nil
 }
 
+// createStorageAccount creates an Azure storage account
 func createStorageAccount(ctx context.Context, accountName string, az *AzureProvider) (s storage.Account, err error) {
 	storageAccountsClient, err := createStorageAccountClient(az.metadata.subscriptionID)
 
@@ -139,6 +144,7 @@ func createStorageAccount(ctx context.Context, accountName string, az *AzureProv
 	return future.Result(storageAccountsClient)
 }
 
+// createStorageAccountClient creates a client to communicate with Azure
 func createStorageAccountClient(subscriptionID string) (storage.AccountsClient, error) {
 	accountClient := storage.NewAccountsClient(subscriptionID)
 	logrus.Info("Authenticating...")
@@ -152,6 +158,7 @@ func createStorageAccountClient(subscriptionID string) (storage.AccountsClient, 
 	return accountClient, nil
 }
 
+// determineParameters determines the access mode from PVC
 func (az *AzureProvider) determineParameters(pvc *v1.PersistentVolumeClaim) (map[string]string, error) {
 	var parameter = map[string]string{}
 	for _, mode := range pvc.Spec.AccessModes {
@@ -171,6 +178,7 @@ func (az *AzureProvider) determineParameters(pvc *v1.PersistentVolumeClaim) (map
 	return nil, errors.New("could not determine parameters")
 }
 
+// determineProvisioner determines what kind of provisioner should the storage class use
 func (az *AzureProvider) determineProvisioner(pvc *v1.PersistentVolumeClaim) (string, error) {
 	for _, mode := range pvc.Spec.AccessModes {
 		switch mode {
@@ -185,10 +193,12 @@ func (az *AzureProvider) determineProvisioner(pvc *v1.PersistentVolumeClaim) (st
 	return "", errors.New("AccessMode is missing from the PVC")
 }
 
+// CheckBucketExistence checks if the bucket already exists
 func (az *AzureProvider) CheckBucketExistence(bucket *v1alpha1.ObjectStore) (bool, error) {
 	return false, nil
 }
 
+// CreateObjectStoreBucket creates a bucket in a cloud specific object store
 func (az *AzureProvider) CreateObjectStoreBucket(*v1alpha1.ObjectStore) error {
 	return nil
 }

--- a/pkg/stub/providers/common.go
+++ b/pkg/stub/providers/common.go
@@ -12,6 +12,7 @@ import (
 	"net/http"
 )
 
+// CommonProvider bonds together the required methods
 type CommonProvider interface {
 	CreateStorageClass(*v1.PersistentVolumeClaim) error
 	GenerateMetadata() error
@@ -21,6 +22,7 @@ type CommonProvider interface {
 	CheckBucketExistence(*v1alpha1.ObjectStore) (bool, error)
 }
 
+// DetermineProvider determines the cloud provider type based on metadata server
 func DetermineProvider() (CommonProvider, error) {
 	var providers = map[string]string{
 		"azure":  "http://169.254.169.254/metadata/instance?api-version=2017-12-01",
@@ -54,6 +56,7 @@ func DetermineProvider() (CommonProvider, error) {
 	return nil, fmt.Errorf("could not determine cloud provider")
 }
 
+// CheckPersistentVolumeClaimExistence checks if the PVC already exists
 func CheckPersistentVolumeClaimExistence(name string) bool {
 	persistentVolumeClaim := &v1.PersistentVolumeClaim{
 		TypeMeta: metav1.TypeMeta{
@@ -73,6 +76,7 @@ func CheckPersistentVolumeClaimExistence(name string) bool {
 	return true
 }
 
+// CheckStorageClassExistence checks if the storage class already exists
 func CheckStorageClassExistence(name string) bool {
 	storageClass := &storagev1.StorageClass{
 		TypeMeta: metav1.TypeMeta{

--- a/pkg/stub/providers/google.go
+++ b/pkg/stub/providers/google.go
@@ -14,10 +14,12 @@ import (
 	"net/http"
 )
 
+// GoogleProvider holds info about Google provider and allows us to implement the common interface
 type GoogleProvider struct {
 	projectId string
 }
 
+// CreateStorageClass creates a StorageClass based on specs described on PVC
 func (gke *GoogleProvider) CreateStorageClass(pvc *v1.PersistentVolumeClaim) error {
 	logrus.Info("Creating new storage class")
 	provisioner, err := gke.determineProvisioner(pvc)
@@ -46,10 +48,12 @@ func (gke *GoogleProvider) CreateStorageClass(pvc *v1.PersistentVolumeClaim) err
 	})
 }
 
+// GenerateMetadata generates metadata which are needed to create a StorageClass
 func (gke *GoogleProvider) GenerateMetadata() error {
 	return nil
 }
 
+// determineParameters determines the access mode from PVC
 func (gke *GoogleProvider) determineParameters(pvc *v1.PersistentVolumeClaim) (map[string]string, error) {
 	//var parameter = map[string]string{}
 	for _, mode := range pvc.Spec.AccessModes {
@@ -61,6 +65,7 @@ func (gke *GoogleProvider) determineParameters(pvc *v1.PersistentVolumeClaim) (m
 	return nil, errors.New("could not determine parameters")
 }
 
+// determineProvisioner determines what kind of provisioner should the storage class use
 func (gke *GoogleProvider) determineProvisioner(pvc *v1.PersistentVolumeClaim) (string, error) {
 	for _, mode := range pvc.Spec.AccessModes {
 		switch mode {
@@ -73,6 +78,7 @@ func (gke *GoogleProvider) determineProvisioner(pvc *v1.PersistentVolumeClaim) (
 	return "", errors.New("AccessMode is missing from the PVC")
 }
 
+// determineProjectId determines the project ID from the metadata server
 func (gke *GoogleProvider) determineProjectId() error {
 	logrus.Info("Getting ProjectID from Metadata service")
 	req, err := http.NewRequest("GET", "http://169.254.169.254/0.1/meta-data/project-id", nil)
@@ -94,6 +100,7 @@ func (gke *GoogleProvider) determineProjectId() error {
 	return nil
 }
 
+// CreateObjectStoreBucket creates a bucket in a cloud specific object store
 func (gke *GoogleProvider) CreateObjectStoreBucket(app *v1alpha1.ObjectStore) error {
 	ctx := context.Background()
 	logrus.Info("Creating new storage client")
@@ -114,6 +121,7 @@ func (gke *GoogleProvider) CreateObjectStoreBucket(app *v1alpha1.ObjectStore) er
 	return nil
 }
 
+// CheckBucketExistence checks if the bucket already exists
 func (gke *GoogleProvider) CheckBucketExistence(app *v1alpha1.ObjectStore) (bool, error) {
 	return false, nil
 }

--- a/pkg/stub/providers/nfs.go
+++ b/pkg/stub/providers/nfs.go
@@ -14,6 +14,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// SetUpNfsProvisioner sets up a deployment a pvc and a service to handle nfs workload
 func SetUpNfsProvisioner(pv *v1.PersistentVolumeClaim) error {
 	logrus.Info("Creating new PersistentVolumeClaim for Nfs provisioner..")
 
@@ -176,6 +177,7 @@ func SetUpNfsProvisioner(pv *v1.PersistentVolumeClaim) error {
 	return nil
 }
 
+// CheckNfsServerExistence checks if the NFS deployment and all companion service exists
 func CheckNfsServerExistence(name string) bool {
 	if !CheckPersistentVolumeClaimExistence(fmt.Sprintf("%s-data", name)) {
 		logrus.Info("PersistentVolume claim for Nfs does not exists!")
@@ -193,6 +195,7 @@ func CheckNfsServerExistence(name string) bool {
 
 }
 
+// checkNfsProviderDeployment checks if the NFS deployment exists
 func checkNfsProviderDeployment() bool {
 	deployment := &v1beta1.Deployment{
 		TypeMeta: metav1.TypeMeta{

